src/lualib/Error.ts

```diff
@@ -0,0 +1,78 @@
+type TSTLCapturedErrorStack = Array<{
+    namewhat: string;
+    name?: string;
+    source: string;
+    short_src: string;
+    currentline: number;
+}>;
+
+function __TS__GetErrorStack(constructor: Function): TSTLCapturedErrorStack {
+    const functionFrames = [];
+    let level = 1;
+    while (true) {
+        const info = debug.getinfo(level, "f");
+        level += 1;
+        if (!info) {
+            // constructor not in call stack
+            level = 1;
+            break;
+        } else if (info.func === constructor) {
+            break;
+        }
+    }
+    while (true) {
+        const info = debug.getinfo(level, "Snl");
+        if (!info) break;
+        if (info.currentline !== -1) {
+            functionFrames.push(info);
+        }
+        level += 1;
+    }
+    return functionFrames;
+}
+
+function __TS__ConvertErrorStack(stack: TSTLCapturedErrorStack): string {
```

Comment:
Imo the error stack should match `debug.getinfo`. As far as I can tell there is also no standard way for users to retrieve the sourcemaps, so it makes sense to directly convert here as well.

Reply:
Also in the future, the call to `__TS__ConvertErrorStack` can be moved to the getter so it won't be invoked during the snapshot.

Reply:
In any case, builtin source map handling is a kind of temporary thing:

> // TODO: In the future, change this to __TS__RegisterFileInfo and provide tstl interface to
> // get some metadata about transpilation.

In the future traceback patch would be packed in a separate module, which would have better module and bundling support and would allow to use sourcemaps for environments that don't call custom `debug.traceback`.

So what is needed there is some extension point that would allow to tap into stack formatting and allow to change it in environment-specific way. But actually, if we want stack to have the same format as `debug.traceback` has can't we just use `debug.traceback`?

Reply:
Yeah more or less how it was in the beginning. I think now it makes sense to actually rollback to `debug.traceback` for printing the stack.